tests/Unit/Core/Form/IdentifiableObject/DataProvider/ProductFormDataProviderTest.php

```diff
@@ -1058,6 +1139,32 @@ private function createProductFeatureValueOptions(array $productData): array
         return $productFeatureValues;
     }
 
+    /**
+     * @param array $productData
+     *
+     * @return PackedProductDetails[]
+     */
+    private function createPackedProductsDetails(array $productData): array
```

Comment:
Nice this should be enough once the query handler handles the image internally (then just mocking the result `PackedProductDetails` - or `PackedProduct` - will be enough)

Reply:
ok I will merge it in once

Reply:
pushed